spec/requests/dashboard_spec.rb

```diff
@@ -300,4 +313,77 @@
       end
     end
   end
+
+  # TODO: (Alex Smith) - update roles before release
+  describe "GET /dashboard/subscriptions" do
+    before do
+      sign_in user
+    end
+
+    it "renders subscriptions" do
+      user.add_role(:admin)
+      article_with_user_subscription_tag = create(:article, user: user, with_user_subscription_tag: true)
+      user_subscription = create(:user_subscription,
+                                 subscriber_id: second_user.id,
+                                 subscriber_email: second_user.email,
+                                 author_id: article_with_user_subscription_tag.user_id,
+                                 user_subscription_sourceable: article_with_user_subscription_tag)
+
+      get "/dashboard/subscriptions", params: { source_type: article_with_user_subscription_tag.class.name, source_id: article_with_user_subscription_tag.id }
+      expect(response.body).to include(user_subscription.subscriber_email)
+    end
+
+    it "displays a message if no subscriptions are found" do
+      get "/dashboard/subscriptions", params: { source_type: article.class.name, source_id: article.id }
+      expect(response.body).to include("You don't have any subscribers for this")
+    end
+
+    it "raises unauthorized when trying to access a source the user doesn't own" do
+      user.add_role(:admin)
+      article_with_user_subscription_tag = create(:article, :with_user_subscription_tag_role_user, with_user_subscription_tag: true)
+      create(:user_subscription,
+             subscriber_id: second_user.id,
+             subscriber_email: second_user.email,
+             author_id: article_with_user_subscription_tag.user_id,
+             user_subscription_sourceable: article_with_user_subscription_tag)
+
+      expect do
+        get "/dashboard/subscriptions", params: { source_type: article_with_user_subscription_tag.class.name, source_id: article_with_user_subscription_tag.id }
```

Comment:
I see this being used a couple times throughout the test; can we pull it out into a variable? This specific `get` is really hard to read/wrap your head around :)

Reply:
It is used a couple of times and there are variations of this in the file where it's slightly different.

What if I did:

```ruby
expect do
  params = {
    source_type: article_with_user_subscription_tag.class.name,
    source_id: article_with_user_subscription_tag.id
  }

  get "/dashboard/subscriptions", params: params
end.to raise_error(Pundit::NotAuthorizedError)
```

Does this help with readability?

Reply:
@atsmith813 Hm, I thought it was the same line that is used 4 times on 332, 351, 374, and 385, which is why I suggested extracting it 🤔

Reply:
@vaidehijoshi that's correct! And 363 and 337 are different. I usually find it more confusing to override an extraction (`params`) only in specific specs (363 and 337) so I usually only extract it out if it's going to be consistent (like putting `user` at the top level).

I'll extract it though and re-name them.

Reply:
I'm also hesitant to extract when the params are defined by something else in the spec (`article_with_user_subscription_tag`). That has a low chance of changing in the future here so it should be fine to take it out as well.

Reply:
@vaidehijoshi On second thought, I'll revisit this in the next PR when I change the specs again to update them for the new roles. The challenge here is sometimes I make the `article_with_user_subscription_tag` differently, depending on the spec. And I do that because of roles.

Reply:
@atsmith813 Ok, that's fair! Seems like it will be easier to change it in the next PR pass through.